packages/agent/src/agent/http/index.ts

```diff
@@ -1126,13 +1146,31 @@ export class HttpAgent implements Agent {
   }
 
   public async fetchRootKey(): Promise<ArrayBuffer> {
```

Comment:
`#rootKeyPromise` is never actually set so the .

```typescript
  public async fetchRootKey(): Promise<ArrayBuffer> {
    if (this.#fetchRootKeyPromise === null) {
      this.#fetchRootKeyPromise = this.#fetchRootKey();
    }

    const result = await this.#fetchRootKeyPromise;
    this.rootKey = result;
    return result;
  }

  async #fetchRootKey(): Promise<ArrayBuffer> {
    const status = await this.status();
    // Hex-encoded version of the replica root key
    const rootKey = (status as JsonObject & { root_key: ArrayBuffer }).root_key;
    this.rootKey = rootKey;

    return rootKey;
  }
```

Reply:
Thanks for catching!